website/docs/language/expressions/type-constraints.mdx

```diff
@@ -89,12 +89,14 @@ The three kinds of collection type in the Terraform language are:
   compatibility with older configurations; for new code, we recommend using
   the full form.
 
-  Maps can be made with braces ({}) and colons (:) or equals signs (=):
-  { "foo": "bar", "bar": "baz" } OR { foo = "bar", bar = "baz" }. Quotes
-  may be omitted on keys, unless the key starts with a number, in which
-  case quotes are required. Commas are required between key/value pairs
-  for single line maps. A newline between key/value pairs is sufficient
-  in multi-line maps.
+  Maps can be made with braces ({}) and colons (:) or equals signs (=): 
+  { "foo": "bar", "bar": "baz" } OR { foo = "bar", bar = "baz" }. 
+  Quotes may be omitted on keys, except in certain situations where they 
+  are necessary, such as when a key starts with a number, contains spaces, 
+  or has special characters. In these cases, quotes are required. 
+  Additionally, for single-line maps, commas are required between 
+  key/value pairs, while in multi-line maps, a newline between 
+  key/value pairs is sufficient.
```

Comment:
```suggestion
  You can use the following characters to define maps: 
      - `{}` 
      - `:` 
      - `=` 
  For example, `{ "foo": "bar", "bar": "baz" }` and  `{ foo = "bar", bar = "baz" }` define the same map. 
  You must place map keys in quotation marks when a key starts with a number, contains spaces, or contains special characters. Otherwise, you can omit them.
  You must place commas between keys-value pairs in single-line maps. 
  You can place key-value pairs in multi-line maps on new lines.
```
Recast in active voice to make the information clearer.